packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -503,7 +614,308 @@ test.serial('writeGranulesFromMessage() saves the same values to DynamoDB, Postg
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    providerCumulusId,
+    knex,
+  } = t.context;
+
+  // Need a message in 'completed' state to allow files writes
+  const completedCumulusMessage = {
+    cumulus_meta: {
+      workflow_start_time: t.context.workflowStartTime,
+      state_machine: t.context.stateMachineArn,
+      execution_name: t.context.executionName,
+    },
+    meta: {
+      status: 'completed',
+      collection: t.context.collection,
+      provider: t.context.provider,
+    },
+    payload: {
+      granules: [granule],
+    },
+  };
+
+  await writeGranulesFromMessage({
+    cumulusMessage: completedCumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const originalDynamoGranule = await granuleModel.get({ granuleId });
+  const originalEsGranule = await esGranulesClient.get(granuleId);
+  const originalpgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+  const originalApiGranule = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: originalpgGranule,
+    knexOrTransaction: knex,
+  });
+
+  const originalPayloadFiles = files;
+
+  originalPayloadFiles.sort(
+    (f1, f2) => sortfilesByBuckets(f1, f2)
+  );
+  originalApiGranule.files.sort(
+    (f1, f2) => sortfilesByBuckets(f1, f2)
+  );
+  originalDynamoGranule.files.sort(
+    (f1, f2) => sortfilesByBuckets(f1, f2)
+  );
+  originalEsGranule.files.sort(
+    (f1, f2) => sortfilesByBuckets(f1, f2)
+  );
+
+  // Files were written correctly in initial DB writes
+  t.deepEqual(originalDynamoGranule.files, originalPayloadFiles);
+  t.deepEqual(originalEsGranule.files, originalPayloadFiles);
+  t.deepEqual(originalApiGranule.files, originalPayloadFiles);
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    files: [],
+    status: granule.status,
+  };
+
+  const {
+    updatedPgGranuleFields,
+    pgGranule,
+    esGranule,
+    dynamoGranule,
+  } = await updateGranule(t, updateGranulePayload, true);
+
+  // Postgres granule matches expected updatedGranule
+  t.deepEqual(
+    omit(removeNilProperties(pgGranule), cumulusMessageOmitList),
+    omit(
+      removeNilProperties({ ...originalpgGranule, ...updatedPgGranuleFields }),
+      cumulusMessageOmitList
+    )
+  );
+
+  const apiGranule = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: pgGranule,
+    knexOrTransaction: knex,
+  });
+
+  // Files were removed from all datastores
+  t.deepEqual(apiGranule.files, []);
+  t.deepEqual(dynamoGranule.files, []);
+  t.deepEqual(esGranule.files, []);
+
+  // Postgres and ElasticSearch granules matches
+  t.deepEqual(
+    apiGranule,
+    omit(esGranule, ['_id'])
+  );
+
+  // Postgres and Dynamo granules matches
+  t.deepEqual(
+    apiGranule,
+    dynamoGranule
+  );
```

Comment:
We maybe don't need to check that the tables are in-sync in this test as we really just care that the files were removed. It seemed like a redundant check wasn't necessarily bad or confusing but open to feedback 🤷

Reply:
Thinking about, I might lean towards keeping the assertions as-is. Explicitly checking that all files are `[]` in all datastores doesn't seem like the worst thing to me given the issues we've seen.